test/Tests/src/Semantic/Conversion_Spec.enso

```diff
@@ -421,6 +422,38 @@ spec =
 
             do_duration now
 
+    Test.group "Polyglot Conversion" <|
+        f1 (x : DateTimeFormatter) = x.to_text
+        f2 (x : Text | DateTimeFormatter) = x.to_text
+        f3 (x : DateTimeFormatter | Text) = x.to_text
+        f4 (x) = case x of
+            _ : DateTimeFormatter -> "DateTimeFormatter: "+x.to_text
+            _ : Text -> "TEXT: "+x
+            _ -> "OTHER"
+
+        d = DateTimeFormatter.ISO_DATE
+
+        Test.specify "f1 d" <|
+            (f1 d).to_display_text . should_contain "ParseCaseSensitive(false)(Value"
+
+        Test.specify "f1 42" <|
+            Panic.recover Any (f1 42).to_display_text . should_fail_with Type_Error
```

Comment:
Preferred to use the method made for this specifically:
```suggestion
            Test.expect_panic_with (f1 42).to_display_text Type_Error
```

Reply:
My code is using the `recover` and `should_fail_with` already at many places.